provider/safedns/safedns.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package safedns
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	ukf_client "github.com/ukfast/sdk-go/pkg/client"
+	ukf_connection "github.com/ukfast/sdk-go/pkg/connection"
+	"github.com/ukfast/sdk-go/pkg/service/safedns"
+
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+	"sigs.k8s.io/external-dns/provider"
+)
+
+// The SafeDNS interface is a subset of the SafeDNS service API that are actually used.
```

Comment:
It’s a convention to have the first word of the comment be the name of the type you are commenting. For example:

```suggestion
// SafeDNS is an interface that is a subset of the SafeDNS service API that are actually used.
```


Can you please make it consistent and make sure all exported types have comments?